client/encryptUtils.js

```diff
@@ -0,0 +1,45 @@
+const crypto = require('crypto')
+const eccrypto = require("eccrypto");
+
+module.exports.encryptPlainTextAndKey = async(host, plaintext, publicKey) => {
+      const symmetricKey = crypto.randomBytes(24)
+      const iv = crypto.randomBytes(16)
+
+      const cipher = crypto.createCipheriv('aes192', symmetricKey, iv)
+      const encryptedMessage = Buffer.concat([cipher.update(plaintext, 'utf-8'), cipher.final()])
+      const publicKeyBuffer = Buffer.concat([
+          Buffer.from('04', 'hex'),
+          Buffer.from(publicKey.replace('0x', ''), 'hex')
+      ])
+
+      const hostNameHash = crypto.createHash('sha256');
+      hostNameHash.update(host);
+      const encryptedSymmetricObj = await eccrypto.encrypt(publicKeyBuffer, Buffer.from(
+          `|${hostNameHash.digest('hex')}|${symmetricKey.toString('hex')}|${iv.toString('hex')}|`
+      ))
+
+      const encryptedSymmetricKey = {}
+
+      for (const k in encryptedSymmetricObj) {
+          encryptedSymmetricKey[k] = encryptedSymmetricObj[k].toString('hex')
+      }
+      return {
+          encryptedMessage: encryptedMessage.toString('hex'),
+          encryptedSymmetricObj,
+          encryptedSymmetricKey: JSON.stringify(encryptedSymmetricKey)
+      }
+  }
+
+module.exports.decryptCipherTextAndKey = async (host, cypertext, encryptedSymmetricObj, privateKey) => {
+    const decryptHostNameHash = crypto.createHash('sha256');
+    decryptHostNameHash.update(host);
+    const symmetricObj = await eccrypto.decrypt(Buffer.from(privateKey, 'hex'), encryptedSymmetricObj)
+    const [, hostNameHash, symmetricKey, iv] = symmetricObj.toString().split('|')
+    if (decryptHostNameHash.digest('hex') !== hostNameHash){
+        return {plaintext: null, hostNameHash: null, symmetricKey: null, iv:null}
```

Comment:
I would better throw an error here. It will be easier to figure out what is actually going on.